Header header
MaskInfo[] masks_info